src/Patcher/GitInitPatcher.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace cweagans\Composer\Patcher;
+
+use Composer\IO\IOInterface;
+use cweagans\Composer\Patch;
+
+class GitInitPatcher extends GitPatcher
+{
+    public function apply(Patch $patch, string $path): bool
+    {
+        // If the target is already a git repo, the standard Git patcher can handle it.
+        if (is_dir($path . '/.git')) {
+            return false;
+        }
+
+        $this->io->write("Creating temporary fake git repo in $path to apply patch", true, IOInterface::VERBOSE);
+
+        // Create a fake Git repo -- just enough to make Git think it's looking at a real repo.
```

Comment:
👋 

I think by not calling `git init` that it relies too much on the internal implementation of the `.git` directory. What if a later version requires some fourth file?

I suppose if this is documented somewhere by git it would be OK.

Reply:
I think you're right. Changed this to just use `git init` in a14cba705c2d61d7ac73bb9ed00576cda11f0f30.